zng/docs/zeek-compat.md

```diff
@@ -1,138 +1,188 @@
+T	123	456	123.4560	1592502151.123456	123.456	smile\xf0\x9f\x98\x81smile	\x09\x07\x04	80	127.0.0.1	10.0.0.0/8	tcp	things,in,a,set	order,is,important	Jeanne	122
+
+$ zq -t zeek_types.log 
+#zenum=string
+#0:record[my_bool:bool,my_count:uint64,my_int:int64,my_double:float64,my_time:time,my_interval:duration,my_printable_string:bstring,my_bytes_string:bstring,my_port:port,my_addr:ip,my_subnet:net,my_enum:zenum,my_set:set[bstring],my_vector:array[bstring],my_record:record[name:bstring,age:uint64]]
+0:[T;123;456;123.456;1592502151.123456;123.456;smile😁smile;\x09\x07\x04;80;127.0.0.1;10.0.0.0/8;tcp;[a;in;set;things;][order;is;important;][Jeanne;122;]]
+
+$ zq -t zeek_types.log | zq -f zeek -
+#separator \x09
+#set_separator	,
+#empty_field	(empty)
+#unset_field	-
+#fields	my_bool	my_count	my_int	my_double	my_time	my_interval	my_printable_string	my_bytes_string	my_port	my_addr	my_subnet	my_enum	my_set	my_vector	my_record.name	my_record.age
+#types	bool	count	int	double	time	interval	string	string	port	addr	subnet	enum	set[string]	vector[string]	string	count
+T	123	456	123.456	1592502151.123456	123.456	smile\xf0\x9f\x98\x81smile	\x09\x07\x04	80	127.0.0.1	10.0.0.0/8	tcp	a,in,set,things	order,is,important	Jeanne	122
 ```
-where `<separator>` is intialized to a space character at the beginning of the
-stream, then overridden by the `#separator` directive.
-
-In the legacy format, the `#separator` character and the `#set_separator` character
-define how to parse both a legacy value line and a legacy descriptor.
-
-Every legacy value line corresponds to a `record` type defined by the
-fields and types directives possibly modified for the `#path` directive
-as described below.
-
-Record types may not be used in the `#types` directive,
-which means there is no need to recursively parse the `set` and `vector`
-container values (`set` and `vector` values are split according to
-the `#set_separator` character).
-
-
-
-## Legacy Values
-
-If a value line begins with `#`, then this
-character must be escaped.
-(Such lines can only exist as legacy values and not regular values
-as regular values always begin with an integer descriptor.)
 
+## Type-Specific Details
+
+As `zq` acts as a reference implementation for ZNG, it's helpful to understand
+the following subtleties of how it reads the following Zeek data types into
+ZNG equivalents and writes them back out again in Zeek log format. Other
+ZNG implementations (should they exist) may handle these differently.
+
+### `double`
+
+As they do not affect accuracy, "trailing zero" decimal digits on Zeek `double`
+values will _not_ be preserved (e.g. `123.4560` becomes `123.456`).
+
+### `set`
+
+Because order within sets is not significant, no attempt is made by `zq` to
+maintain the order of `set` elements as they originally appeared in a Zeek log.
+
+### `enum`
+
+As they're encountered in common programming languages, enum variables
+typically hold one of a set of predefined constants. While this is
```

Comment:
```suggestion
typically hold one of a set of predefined values. While this is
```

Reply:
@henridf: I actually used the word "constants" in this way three times in the paragraph but you only flagged the first one. I assume you'd advocate making the change in all places?

FWIW, I only chose this lingo because the top Google hit on "enum type" spoke about it this way (admittedly it was a Java ref), but I'm perfectly happy to flip the wording. I just wanted to make certain there wasn't subtle reason why _this_ ref should change from "constants" to "values" but the other two were still seen as appropriate for some reason.

Reply:
There is no reason I just suggested changing this one and not the others - I probably just missed the others. And if you've found other wording that uses "constant", it's entirely ok with me to drop this suggestion.

Reply:
Thanks @henridf. I went ahead and standardized on "constants", then. You got me to thinking that "constant" is a word that comes up more in programming language contexts, and in this case we're closer to a log/query context, so "values" does, indeed, feel more appropriate.